### PR TITLE
feat(toInteger32): add toInteger32

### PR DESCRIPTION
**Pull request by OliverRadini**
#783

### CODE REVIEW

File: src/toInteger32.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { curryN } from 'ramda';
+/**
+ * Converts double-precision 64-bit binary format IEEE 754 to signed 32 bit integer number.
+ *
+ * @func toInteger32
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/1.12.0|v1.12.0}
```
Comment (char0n) [inline src/toInteger32.js:7]:
```suggestion
 * @since {@link https://char0n.github.io/ramda-adjunct/2.28.0|v2.28.0}
```

File: src/toInteger32.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { curryN } from 'ramda';
+/**
+ * Converts double-precision 64-bit binary format IEEE 754 to signed 32 bit integer number.
+ *
+ * @func toInteger32
+ * @memberOf RA
```
Comment (char0n) [inline src/toInteger32.js:6]:
```suggestion
 * @aliases toInt32
 * @memberOf RA
```

File: src/toInteger32.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * @since {@link https://char0n.github.io/ramda-adjunct/1.12.0|v1.12.0}
+ * @category Function
+ * @sig Number -> Number
+ * @param {Number} number A number
+ * @return {Number} A 32-bit integer number
+ * @see {@link http://speakingjs.com/es5/ch11.html#integers_via_bitwise_operators}
+ * @example
+ *
+ * const n = Math.pow(2, 32) + 1;
+ * const as32BitInteger = toInteger32(n); // => 1
```
Comment (char0n) [inline src/toInteger32.js:16]: I'd replace the example with the following one:
```js
RA.toInteger32(2 ** 35); // => 0
RA.toInteger32(2 ** 30); // => 1073741824
```

File: test/toInteger32.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import * as R from 'ramda';
+import { assert } from 'chai';
+
+import * as RA from '../src';
+
+describe('toInteger32', function () {
+  context('given a small integer', function () {
+    specify('should maintain the same value', function () {
+      assert.strictEqual(RA.toInteger32(1 + Math.random()), 1);
```
Comment (char0n) [inline test/toInteger32.js:9]: Please don't use `Math.random` in tests. Use numbers constants like `1` or `200` to make the test explicit and clear. This is not `property based test` so randomization of values in not something we want to do here.

Comment (char0n) [reply test/toInteger32.js:9]: This applies for all the tests in this file.

File: test/toInteger32.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+      assert.strictEqual(RA.toInteger32(new Error()), 0);
+      assert.strictEqual(RA.toInt32(new Error()), 0);
+    });
+  });
+
+  context('given Infinity value', function () {
+    specify('should return 0', function () {
+      assert.strictEqual(RA.toInteger32(Infinity), 0);
+      assert.strictEqual(RA.toInt32(Infinity), 0);
```
Comment (char0n) [inline test/toInteger32.js:63]: Let's also add tests for `-Infinity`.
```suggestion
      assert.strictEqual(RA.toInt32(Infinity), 0);
      assert.strictEqual(RA.toInt32(-Infinity), 0);
      assert.strictEqual(RA.toInt32(-Infinity), 0);      
```

File: test/toInteger32.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    });
+  });
+
+  it('should support placeholder to specify "gaps"', function () {
+    const f0 = RA.toInteger32(R.__);
+    const f1 = RA.toInt32(R.__);
+
+    assert.strictEqual(f0(1.2), 1);
```
Comment (char0n) [inline test/toInteger32.js:71]: Please don't test an alias here. This test should test the curring not the alias. For aliases we have a specialized [convention](https://github.com/char0n/ramda-adjunct/blob/b95fa04adaed288259947dbac5f0225e0ae138f4/test/seq.js#L44) how to test them. Please use the contention here.

File: src/toInteger32.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { curryN } from 'ramda';
+/**
+ * Converts double-precision 64-bit binary format IEEE 754 to signed 32 bit integer number.
+ *
+ * @func toInteger32
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/1.12.0|v1.12.0}
+ * @category Function
```
Comment (char0n) [inline src/toInteger32.js:8]:
```suggestion
 * @category Math
```

File: src/index.js [hunk 8]
```diff
@@ -204,6 +204,8 @@ export { default as floor } from './floor';
 export { default as trunc } from './trunc';
 export { default as sign } from './sign';
 export { default as subtractNum } from './subtractNum';
+export { default as toInteger32 } from './toInteger32';
+export { default as toInt32 } from './toInteger32';
```
[the remark this answers is not in the document]
Comment (OliverRadini) [reply src/index.js:208]: 👍

File: src/toInteger32.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+ * @sig Number -> Number
+ * @param {Number} number A number
```
[the remark this answers is not in the document]
Comment (OliverRadini) [reply src/toInteger32.js:10]: 👍

File: src/toInteger32.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+ * @return {Number} A 32-bit integer number
```
[the remark this answers is not in the document]
Comment (OliverRadini) [reply src/toInteger32.js:11]: 👍

File: src/toInteger32.js (hunk 1 above)
Comment (OliverRadini) [reply src/toInteger32.js:7]: Thanks, I was unsure how to populate this one

File: src/toInteger32.js (hunk 2 above)
Comment (OliverRadini) [reply src/toInteger32.js:6]: 👍

File: src/toInteger32.js (hunk 3 above)
Comment (OliverRadini) [reply src/toInteger32.js:16]: 👍

File: test/toInteger32.js (hunk 4 above)
Comment (OliverRadini) [reply test/toInteger32.js:9]: Fair enough if you don't want randomisation - I'll change it

File: test/toInteger32.js (hunk 5 above)
Comment (OliverRadini) [reply test/toInteger32.js:63]: 👍

File: test/toInteger32.js (hunk 6 above)
Comment (OliverRadini) [reply test/toInteger32.js:71]: All of the tests in the file test the alias - should they all be removed?

File: src/toInteger32.js (hunk 7 above)
Comment (OliverRadini) [reply src/toInteger32.js:8]: 👍

**Comment by OliverRadini** [comment]
I believe all the requested changes are now made and commited

**Comment by char0n** [comment]
@OliverRadini next time you can directly incorporate my suggestions via github: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

File: src/toInteger32.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * @func toInteger32
+ * @aliases toInt32
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.28.0|v2.28.0}
+ * @category Math
+ * @sig Number -> Number
+ * @param {number} number A number
+ * @return {number} A 32-bit integer number
+ * @see {@link http://speakingjs.com/es5/ch11.html#integers_via_bitwise_operators}
```
Comment (char0n) [inline src/toInteger32.js:13]:
```suggestion
 * @see {@link RA.toUInteger32|toUInteger32}, {@link http://speakingjs.com/es5/ch11.html#integers_via_bitwise_operators}
```

File: test/toInteger32.js (hunk 6 above)
Comment (char0n) [reply test/toInteger32.js:53]: Yes we should only test original function, and then have one additional function for alias. But looking at code, you already did that.

File: types/test/toInteger32.ts [hunk 12]
```diff
@@ -0,0 +1,3 @@
+import * as RA from 'ramda-adjunct';
+
+RA.toInteger32(6);
```
Comment (char0n) [inline types/test/toInteger32.ts:3]:
```suggestion
RA.toInteger32(1); // $ExpectType number
RA.toInteger32(Infinity); // $ExpectType number
RA.toInteger32(-Infinity); // $ExpectType number
RA.toInteger32(+Infinity); // $ExpectType number
RA.toInteger32(NaN); // $ExpectType number
RA.toInteger32({}); // $ExpectError
RA.toInteger32('s'); // $ExpectError
RA.toInteger32([]); // $ExpectError

RA.toInt32(1); // $ExpectType number
RA.toInt32('s'); // $ExpectError
```

File: test/toInteger32.js [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+  context('given Infinity value', function () {
+    specify('should return 0', function () {
+      assert.strictEqual(RA.toInteger32(Infinity), 0);
+      assert.strictEqual(RA.toInteger32(-Infinity), 0);
+    });
+  });
+
+  it('should support placeholder to specify "gaps"', function () {
+    const f0 = RA.toInteger32(R.__);
```
Comment (char0n) [inline test/toInteger32.js:51]:
```suggestion
    const toInteger32 = RA.toInteger32(R.__);
```

File: test/toInteger32.js [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+    assert.strictEqual(f0(1.2), 1);
```
Comment (char0n) [inline test/toInteger32.js:53]:
```suggestion
    assert.strictEqual(toInteger32(1.2), 1);
```

File: test/toInteger32.js [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+    specify('should maintain the same value', function () {
+      assert.strictEqual(RA.toInteger32(1), 1);
+    });
+
+    specify('should preserve the sign', function () {
+      assert.strictEqual(RA.toInteger32(-1), -1);
+    });
+  });
+
+  context('given an integer 1 modulo 23^2', function () {
```
Comment (char0n) [inline test/toInteger32.js:17]:
```suggestion
  context('given an integer 23^2 + 1', function () {
```

File: test/toInteger32.js [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+    specify('should give one when converted', function () {
```
Comment (char0n) [inline test/toInteger32.js:18]:
```suggestion
    specify('should return 1', function () {
```

File: test/toInteger32.js [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+  });
+
+  context('given an integer 23^2 + 1', function () {
+    specify('should return 1', function () {
+      const initialValue = 2 ** 32 + 1;
+      assert.strictEqual(RA.toInteger32(initialValue), 1);
+    });
+  });
+
+  context('given an integer -1 modulo 23^2', function () {
```
Comment (char0n) [inline test/toInteger32.js:24]:
```suggestion
  context('given an integer 23^2 - 1', function () {
```

File: test/toInteger32.js [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+    specify('should give one when converted', function () {
```
Comment (char0n) [inline test/toInteger32.js:25]:
```suggestion
    specify('should return -1', function () {
```

File: src/toInteger32.js [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+ * Converts double-precision 64-bit binary format IEEE 754 to signed 32 bit integer number.
+ *
+ * @func toInteger32
+ * @aliases toInt32
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.28.0|v2.28.0}
+ * @category Math
+ * @sig Number -> Number
+ * @param {number} number A number
+ * @return {number} A 32-bit integer number
```
Comment (char0n) [inline src/toInteger32.js:12]:
```suggestion
 * @return {number} A signed 32-bit integer number
```

**Comment by OliverRadini** [comment]
@char0n Glad you liked it - thanks for the help getting it to a mergable state.

**Pull request merged**